src/components/shared/globalTypes.ts

```diff
@@ -195,7 +211,31 @@ export class Directory extends FileSystemObject {
     return currentFsObject;
   }
 
+  isEmpty(): boolean {
+    return this.children === undefined || this.children.size === 0;
+  }
+
   private checkHidden(name: string) {
     return name.startsWith('.');
   }
 }
+
+export function getFSObjectHelper(
+  path: string,
+  root: Directory,
+  cwd: Directory,
+  onFileNotFound?: () => string,
+  onIsNotDirectory?: () => string
```

Comment:
Break these up into separate helpers
- `fileNotFoundHelper(onFileNotFound: () => string)`
- `fileIsNotDirectoryHelper(onIsNotDirectory: () => string)`
And then call the master helper function on the inside. This way we enforce the `null` check without having to explicitly check for it

Reply:
Here
```js
interface ErrorObj {
  onFileNotFound?: () => string;
  onIsNotDirectory?: () => string;
}

export function getFSObjectMasterHelper(
  path: string,
  root: Directory,
  cwd: Directory,
  errorObj?: ErrorObj,
): Directory | File | string {
  const fsObject = path.startsWith('/')
    ? root.getFileSystemObjectFromPath(path)
    : cwd.getFileSystemObjectFromPath(path);

  // Return fsObject in the case of nullish or empty errorObj
  if (!errorObj || Object.keys(errorObj).length === 0) {
    return fsObject;
  }

  if (!fsObject && errorObj.onFileNotFound) {
    return errorObj.onFileNotFound()
  }

  if (errorObj.onIsNotDirectory && !fsObject?.isDirectory) {
    return errorObj.onIsNotDirectory();
  }

  // So technically, it's impossible to hit this case
  // but if they do, let's just throw an exception just in case
  throw new Error("I am not sure what you're doing here tbh, please double check?")
}
```

Reply:
I see, yeah this makes sense. I like the idea of using a new interface to handle errors. I will include this in a constants file. Also, will make the change to getFileSystemObjectFromPath to return null rather than undefined, which should work with `!fsObject`.

Reply:
Oh, but instead of throwing the mysterious Error, could just move the return fsObject there and reorder the second if statement to have `!fsObject?.isDirectory` first for short-circuiting. Then, it can only reach the bottom for errors.

Reply:
For the case of !fsObject but no onFileFound, this would return the null. But I guess it's okay because that means we wouldn't need to handle noFIleFound for some reason. Actually, now that I think about it, this might be a non-optional param.